core/domain/LDAStringClassifierTest.py

```diff
@@ -231,3 +232,85 @@ def test_predict_label_for_doc(self):
         labels = self.classifier.predict(self._EXAMPLES_TEST)
         self.assertEquals(labels[2], '_default')
         self._validate_instance()
+
+    def test_validate(self):
+        """Test to verify validate method of LDAStringClassifier."""
+
+        # Verify no errors are raised for correct data.
+        classifier_data = {
+            '_alpha': 0.1,
+            '_beta': 0.001,
+            '_prediction_threshold': 0.5,
+            '_training_iterations': 25,
+            '_prediction_iterations': 5,
+            '_num_labels': 10,
+            '_num_docs': 12,
+            '_num_words': 20,
+            '_label_to_id': {'text': 1},
+            '_word_to_id': {'hello': 2},
+            '_w_dp': [],
+            '_b_dl': [],
+            '_l_dp': [],
+            '_c_dl': [],
+            '_c_lw': [],
+            '_c_l': []
+        }
+        self.classifier.validate(classifier_data)
+
+        # Verify validation error is raised when key is not present.
+        classifier_data.pop('_alpha', None)
+        with self.assertRaisesRegexp(utils.ValidationError, (
+            'Expected _alpha to be a key in classifier_data')):
```

Comment:
Indent this by 4 more, to help distinguish it from the next line. Otherwise they look like they're in the same block.

Ditto elsewhere.